app/src/main/java/com/breens/searchableexposeddropdownmenujetpackcompose/MainActivity.kt

```diff
@@ -55,26 +35,57 @@ class MainActivity : ComponentActivity() {
                         .fillMaxSize()
                         .padding(16.dp)
                 ) {
-
                     val sports =
                         mutableListOf(
-                            "Basketball",
-                            "Rugby",
-                            "Football",
-                            "MMA",
-                            "Motorsport",
-                            "Snooker",
-                            "Tennis"
+                            Test("Name", 1),
+                            Test("Name", 1),
+                            Test("Name", 1),
+                            Test("Name", 1),
+                            Test("Name", 1),
+                            Test("Name", 1)
                         )
-                    SearchableExpandedDropDownMenu(
-                        listOfItems = sports,
-                        modifier = Modifier.fillMaxWidth(),
-                        onDropDownItemSelected = { item ->
-                            Toast.makeText(applicationContext, item, Toast.LENGTH_SHORT).show()
+                    LazyColumn() {
+                        items(50) {
+                            SearchableExpandedDropDownMenu(
+                                listOfItems = sports,
+                                modifier = Modifier.fillMaxWidth(),
+                                onDropDownItemSelected = { item ->
+                                    Toast.makeText(applicationContext, item.name, Toast.LENGTH_SHORT).show()
+                                },
+                                dropdownItem = { test ->
+                                    DropDownItem(test = test)
+                                }
+                            )
+                            Spacer(modifier = Modifier.height(12.dp))
                         }
-                    )
+                    }
                 }
             }
         }
     }
 }
+
+@Composable
+fun DropDownItem(test: Test) {
+    Row(
+        modifier = Modifier
+            .padding(8.dp)
+            .wrapContentSize()
+    ) {
+        Icon(
+            Icons.Outlined.Check,
+            contentDescription = "drop down arrow"
+        )
```

Comment:
I don't think the user will need an icon here, you can put it to be optional

Reply:
They actually have an option. The parameter {dropdownItem} is a higher order that takes a composable as a dropdown item that a user has designed.